### ISSUE TITLE
About the _fix_enc_hidden() function

### ISSUE THREAD

**Issue by lan2720** [OpenNMT/OpenNMT-py#61]
Hi, this project is great.
I found this code
```
    def _fix_enc_hidden(self, h):
        #  the encoder hidden is  (layers*directions) x batch x dim
        #  we need to convert it to layers x batch x (directions*dim)
        if self.encoder.num_directions == 2:
            return h.view(h.size(0) // 2, 2, h.size(1), h.size(2)) \
                    .transpose(1, 2).contiguous() \
                    .view(h.size(0) // 2, h.size(1), h.size(2) * 2)
        else:
            return h
```
If we do this, we assume the **h_n** is like
[
layer0_**forward**
layer0_backward
layer1_**forward**
layer1_backward
layer2_**forward**
layer2_backward
...
]
So how can I know it's correct instead of like 
[
layer0_**forward**
layer1_**forward**
layer2_**forward**
layer0_backward
layer1_backward
layer2_backward
...
]

**Comment by voidmagic** [comment]
Hey, I also confused with question last night, but now I have figured out how it works.

Like this example:
```
torch.manual_seed(1)
lstm = nn.LSTM(input_size=1,
               num_layers=1,
               bidirectional=True,
               bias=False,
               hidden_size=1)

a = Variable(torch.ones(1, 1, 1))
result, context = lstm(a)
print(context)
output: 
Variable containing:
(0 ,.,.) = 
  0.1321

(1 ,.,.) = 
 -0.0980
[torch.FloatTensor of size 2x1x1]
```

and then change its layer to 2:
```
torch.manual_seed(1)
lstm = nn.LSTM(input_size=1,
               num_layers=2,
               bidirectional=True,
               bias=False,
               hidden_size=1)

a = Variable(torch.ones(1, 1, 1))
result, context = lstm(a)
print(context[0])
output: 
Variable containing:
(0 ,.,.) = 
  0.1321

(1 ,.,.) = 
 -0.0980

(2 ,.,.) = 
  0.0037

(3 ,.,.) = 
 -0.0032
[torch.FloatTensor of size 4x1x1]
```
Because I manually set the random seed, so the parameter of LSTM will be same at each time, and the result shows that the order of h is :
[
layer0_forward
layer0_backward
layer1_forward
layer1_backward
layer2_forward
layer2_backward
...
]
And the same for c.

**Comment by voidmagic** [comment]
And I pulled a new request for it just now, see #63 
I rewrite it as `h = torch.cat([h[0:h.size(0):2], h[1:h.size(0):2]], 2)`, it is 10% faster on my computer.

**Comment by lan2720** [comment]
@voidmagic Great! Thanks.
